### PR TITLE
#843 reproducer and solution

### PR DESCRIPTION
**Pull request by sjaakd**
I can't really make a reproducer for this one. You can't test for ineffective code. So I'm wandering whether the UT makes sense.

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_843/Issue843Test.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    GitlabTag.class,
+    TagMapper.class
+})
+@IssueKey("843")
+public class Issue843Test {
+
+    @Test
+    public void testMapperCreation() {
+        TagMapper.INSTANCE.gitlabTagToTagInfo( null );
+        // check the implemenation for ineffective code (duplication of forged method call).
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_843/Issue843Test.java]: You could use the `GeneratedSource` rule to obtain a `JavaFileAssert` with a `.content()` assert on the source code where you can use `containsOnlyOnce` to check that the invocation of the forged method is there only once. Or extend `JavaFileAssert` to do something else based on the generated file if you need it.

Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_843/Issue843Test.java]: Another possibility is to add a counter in the getters and verify that they are not called more than expected.

**Comment by agudian** [comment]
Always interesting to see how simple your fixes look like... :+1:.

I only have a suggestion on how to test it. And reword the commit message as well, please... :smile:

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_843/Issue843Test.java (hunk 1 above)
Comment (sjaakd) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_843/Issue843Test.java]: Of course.. I'll go for the latter one.

**Pull request merged**